addons/udes_stock_routing/models/stock_picking.py

```diff
@@ -104,37 +93,64 @@ def _get_two_stage_configuration_move_mapping(self):
                     operation_type_id=config_location.u_two_stage_intermediate_operation_type.id
                     or self.picking_type_id.id,
                 )
-            ] |= move_line.move_id
+            ] |= move_line
 
-        return configuration_move_mapping
+        return configuration_move_line_mapping
+
+    def split_move_lines_to_backorder(self, move_lines, **kwargs):
+        """
+        Loosely based on backorder_move_lines, but useful when we know the move lines we want to split
+        to a backorder rather than the lines to keep (kind of the inverse of what that function does).
+        """
+        Move = self.env["stock.move"]
+        new_moves = Move.browse()
+        for move_line in move_lines:
+            new_moves |= move_line.move_id.split_out_move_lines(move_line)
+        if new_moves:
+            return self._create_backorder_picking(new_moves, **kwargs)
+        return self.browse()
 
     def initiate_two_stage_split(self):
         """
         Perform all setup to turn a one stage outbound picking to potentially multiple two stage pickings,
-        as per configuration of the reserved moves locations.
+        as per configuration of the reserved move lines locations.
         """
         self.ensure_one()
         Picking = self.env["stock.picking"]
 
         original_operation_type = self.picking_type_id
-        original_location = self.location_id
         original_priority = self.priority
         original_origin = self.origin
-        configuration_move_mapping = self._get_two_stage_configuration_move_mapping()
-        # Unreserve stock on the current picking before proceeding.
-        self.do_unreserve()
-        for configuration, moves in configuration_move_mapping.items():
-            stage_1_pick = Picking.create(
+
+        configuration_move_line_mapping = self._get_two_stage_configuration_move_line_mapping()
+        # # Unreserve stock on the current picking before proceeding.
```

Comment:
will remove this and the below line, not needed.

Reply:
https://github.com/unipartdigital/udes-open/pull/871/commits/840052373340763e332992d572f21826c5eafcd3